### PR TITLE
fix TestTaskCleanupDoesNotDeadlock unix test.

### PR DESCRIPTION
**Pull request by jhaynes**
### Summary
This cleans up the TestTaskCleanupDoesNotDeadlock unix functional test similarly to what #680 did for the equivalent windows test.

### Testing

- [x] Builds on Linux (make release)
- [x] Builds on Windows (go build -out amazon-ecs-agent.exe ./agent)
- [x] Unit tests on Linux (make test) pass
- [x] Unit tests on Windows (go test -timeout=25s ./agent/...) pass
- [x] Integration tests on Linux (make run-integ-tests) pass
- [x] Integration tests on Windows (.\scripts\run-integ-tests.ps1) pass
- [x] Functional tests on Linux (make run-functional-tests) pass
- [x] Functional tests on Windows (.\scripts\run-functional-tests.ps1) pass

### Licensing
This contribution is under the terms of the Apache 2.0 License

### CODE REVIEW

File: agent/functional_tests/tests/functionaltests_unix_test.go [hunk 1]
```diff
@@ -113,12 +113,15 @@ func TestTaskCleanupDoesNotDeadlock(t *testing.T) {
 
 	// Run two Tasks after cleanup, as the deadlock does not consistently occur after
 	// after just one task cleanup cycle.
-	for i := 0; i < 3; i++ {
+	for i := 0; i < 2; i++ {
```
Comment (samuelkarp) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: Why did you shorten this?

File: agent/functional_tests/tests/functionaltests_unix_test.go [hunk 2]
```diff
@@ -127,6 +130,9 @@ func TestTaskCleanupDoesNotDeadlock(t *testing.T) {
 		dockerId, err := agent.ResolveTaskDockerID(testTask, "1")
 		require.NoError(t, err, fmt.Sprintf("Cycle %d: Error resolving docker id for container in task", i))
 
+		err = testTask.Stop()
```
Comment (samuelkarp) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: The original deadlock that got introduced involved short-running containers that stopped themselves.  I think that by making this change you're no longer exercising the codepath that was deadlocking.

File: agent/functional_tests/tests/functionaltests_unix_test.go (hunk 1 above)
Comment (jhaynes) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: To match the comment and the equivalent windows test. However, we can switch to three tasks if we'd rather.

File: agent/functional_tests/tests/functionaltests_unix_test.go (hunk 2 above)
Comment (jhaynes) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: If that is the case, we should scrap this change and update the windows test to be reflective of the same code path.

Comment (aaithal) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: The functional test was added to guard against regressions in task manager's `cleanupTask` method. Specifically to ensure that the task engine ignores/handles any task when it's cleaning a task. This should really be a unit/integ test and doesn't warrant a functional test IMO as it's hard to force that race condition to surface. Especially since the task def being used by the functional test is leading to false negatives most of the time.

In short, yes. @samuelkarp is right when he says 
>  I think that by making this change you're no longer exercising the codepath that was deadlocking.

**Comment by jhaynes** [comment]
Closing in lieu of #752
